src/pudl_archiver/archivers/eia/eiaapi.py

```diff
@@ -0,0 +1,38 @@
+"""Download all bulk data files from EIA API."""
+
+import re
+from pathlib import Path
+
+from pudl_archiver.archivers.classes import (
+    AbstractDatasetArchiver,
+    ArchiveAwaitable,
+    ResourceInfo,
+)
+
+
+class EiaApiFileArchiver(AbstractDatasetArchiver):
+    """EIA bulk file archiver."""
+
+    name = "eiaapi"
+
+    async def get_resources(self) -> ArchiveAwaitable:
+        """Download all EIA bulk API resources."""
+        # Query manifest file to get all existing bulk datasets
+        bulk_manifest = await self.get_json(
+            "https://www.eia.gov/opendata/bulk/manifest.txt"
+        )
+        for dataset in bulk_manifest["dataset"]:
+            # Assign dataset name as partition and file name
+            dataset_name = bulk_manifest["dataset"][dataset]["title"].lower()
+            dataset_name = re.sub("[^0-9a-zA-Z ]+", "", dataset_name).replace(" ", "-")
+            dataset_url = bulk_manifest["dataset"][dataset]["accessURL"]
+            yield self.get_bulk_resource(dataset_name=dataset_name, url=dataset_url)
```

Comment:
this could be a touch less crowded with `for dataset in bulk_manifest["dataset"].values()` but I don't feel strongly about it